### ISSUE TITLE
deadloop on replaying kernel when profiling DL network with nvprof 

### ISSUE THREAD

**Issue by SimonTao** [tensorflow/tensorflow#11070]
Please go to Stack Overflow for help and support:

http://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug or a feature request.
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
Yes, use models from Keras ,e.g. VGG16
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**:
Binary
- **TensorFlow version (use command below)**:
1.1.0
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**:
8.0/5.1
- **GPU model and memory**:
Nvidia K40, 10G
- **Exact command to reproduce**:
nvprof -metrics flop_sp_efficiency python train_vgg
You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

The profiling process will end with a deadloop in replaying some kernels with prompt:
"Replaying kernel "cgemm_sm35_ldg_tn_64x8x64x16x16"

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by ali01** [comment]
@poxvoculi, do we support nvprof?

**Comment by poxvoculi** [comment]
I believe it's possible to use nvprof with TF, but I haven't used it in a long time.  Probably @zheng-xq is better informed.

**Comment by andrei-pokrovsky** [comment]
I hit the same issue - worked around by moving out the actual CUDA kernel into a PyCUDA wrapper and profiling that along with captured launch parameters. I think what's happening is there's some startup overhead in TensorFlow that ends up getting captured by the CUDA profiler unnecessarily (and there's a mountain of it). Another thing to try could be to instrument using the CUDA profiler start/stop API *inside* tensorflow code, because if you use the API from python it has to be inside a session. Something else you could try is to make py_func ops using Python CUDA profiler start/stop API (an idea I just came up with right now, haven't tried it yet)

import ctypes

_cudart = ctypes.CDLL('libcudart.so')

def cu_prof_start():
    ret = _cudart.cudaProfilerStart()
    if ret != 0:
        raise Exception('cudaProfilerStart() returned %d' % ret)


def cu_prof_stop():
    ret = _cudart.cudaProfilerStop()
    if ret != 0:
        raise Exception('cudaProfilerStop() returned %d' % ret)

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
A member of the TensorFlow organization has replied after the stat:awaiting tensorflower label was applied.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 141 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 156 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 171 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 186 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 201 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by linrio** [comment]
Have anyone know how to use the nvprof in tensorflow?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 35 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @zheng-xq: It has been 50 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 65 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by harshini-gadige** [comment]
@linrio Hi, please go through [this link ](https://docs.nvidia.com/cuda/profiler-users-guide/index.html)which gives a picture on using nvprof.

@zheng-xq  Hi, could you please look into this issue ?
